sklearn/metrics/ranking.py

```diff
@@ -95,8 +106,16 @@ def auc(x, y, reorder=False):
             if np.all(dx <= 0):
                 direction = -1
             else:
-                raise ValueError("Reordering is not turned on, and "
-                                 "the x array is not increasing: %s" % x)
+                # Output the most positive value and most negative value in
+                # np.diff(x) for debugging.
```

Comment:
Is it meant to handle the case that adjacent values are out of order but only within a small tolerance?